golos.publication/golos.publication.cpp

```diff
@@ -267,6 +191,47 @@ void publication::close_post_timer() {
     trx.send(_self, _self);
 }
 
+void publication::set_vote(account_name voter, account_name author,
+                           std::string permlink, asset weight,
+                           std::vector<structures::rshares> rshares) {
+    tables::vote_table vote_table(_self, author);
+
```

Comment:
move rshares initialization inside this function, there no need to duplicate it inside `upvote` and `downvote`